esp/smc/SMCLib/TpContainer.cpp

```diff
@@ -743,11 +743,36 @@ static void refreshValidTargets()
     }
 }
 
+static void addUniqueDataPlaneName(const char * dataPlaneName)
+{
+    if (isEmptyString(dataPlaneName))
+        return;
+
+    if (validDataPlaneNames.find(dataPlaneName) == validDataPlaneNames.end())
+    {
+        validDataPlaneNames.insert(dataPlaneName);
+        PROGLOG("adding valid data plane name: %s", dataPlaneName);
+    }
+}
+
+static void refreshValidDataPlaneNames()
+{
+    validDataPlaneNames.clear();
+    Owned<IPropertyTreeIterator> queues = getComponentConfigSP()->getElements("queues");
```

Comment:
I'm not sure what this code is here for.  I think all the planes should be covered by the second iterator.  Did you find an example that wasn't?

Reply:
I did not. But, I saw similar code from the existing getRoxieDefaultPlane() in TpWrapper.cpp. Please let me know if I should remove this code.

Reply:
That code is doing something different - it is finding out what the default data plane associated with a roxie is.
First it checks if the roxie has a specific data plane associated with it.  If not it picks the first data plane.  I suspect with recent changes that should be implemented slightly differently.

Reply:
I will remove the code.